compiler/ilgen/IlBuilder.cpp

```diff
@@ -1363,6 +1363,14 @@ IlBuilder::Div(TR::IlValue *left, TR::IlValue *right)
    return returnValue;
    }
 
+TR::IlValue *
+IlBuilder::Rem(TR::IlValue *left, TR::IlValue *right)
+   {
+   TR::IlValue *returnValue=binaryOpFromOpMap(TR::ILOpCode::remainderOpCode, left, right);
```

Comment:
This looks reasonable (it's consistent with implementations of the other arithmetic services).

But none of our code generators currently support brem and srem, so anyone who uses those types will hit an assert deeper into a compilation than probably desirable. So, maybe we could do an automatic widening here if either of the parameters are Int8 or Int16 (similar but not quite the same as the widening that's done as part of Call for parameters, for example)? 

```
if (left->getDataType() == TR::Int8 || left->getDataType() == TR::Int16)
   left = ConvertTo(Int32, left);
```

...and same for right. Or, even better, you could create a private service `TR::IlValue *widenIntegerTo32Bits(TR::IlValue *value)` to convert both left and right.

For consistency (and in case we revert this widening operation later), I suggest converting the return value to the type of the (arbitrary, but consistent with how other operations choose their type) left operand.

Please also put a comment explaining why the widening is done here, so there's a reminder to people to undo this stuff if/when someone gets around to implementing the opcodes natively.

Reply:
Should be fixed now